Feature/3802 shared feed organizer manage invitations

amoedoamorim opened the pull request:
## Description

This is the "send invitation" feature for shared feeds as per CV2-3802 and its figma design.
Feed organizer (team admin) can invite other teams via email to join a shared feed during Feed creation or editing.

References: CV2-3802

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Tested manually inviting collaborations on feed creation and editing.  Added unit tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

amoedoamorim commented:
@brianfleming I'm still working out a few details here, but could please inspect the styling work already?

brianfleming commented on src/app/components/feed/FeedCollaboration.js:
```diff
@@ -0,0 +1,164 @@
+        />
+      </span>
+      { !feed.dbid &&
+        <FormattedMessage
```
can you wrap this with a class of `typography-body2` so the font sizes to match the designs?

brianfleming commented:
@amoedoamorim no problem. I will just write down some random things I see, that are probably on your list, so ignore if you are on it already. As I test I will add other comments. 

FYI, i just pushed up some tweaks to the icons.

- [ ] after you hit the + button to add the invited email address, the address remains in the textfield, it should clear it out
- [ ] based on your comment the other day, lets update the placeholder text to `Add collaborator email address` 
- [ ] I forgot about the update shared feed confirmation modal, I don't think its necessary to have 2 modals, one to confirm the update, and the other to confirm the invite, can we combine them into one that has both messages when saving a shared feed?
- [ ] I updated the icons to be more generic, but that means you need to add the color for the `person_add.svg` via css instead of it being defined in the svg iteslf
- [ ] I tested adding an email address to invite, adding another, then removing the first. When I tried to save the shared feed, it still thinks I am inviting the first address. See the screenshot below, and look at the email addresses in the modal, and the ones in the right side

<img width="1179" alt="Screenshot 2023-10-27 at 11 25 22 AM" src="https://github.com/meedan/check-web/assets/418001/86215753-88d9-4377-9c4f-7ad7a2a6d680">

brianfleming commented on src/app/components/feed/SaveFeed.module.css:
```diff
@@ -18,10 +18,14 @@
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+  max-width: 350px;
```
the more "correct" flexbox-y way of doing this, would be to remove the max and min widths and go with `flex: 0 0 350px;`
which translates to, don't grow, don't shrink, be 350px wide

brianfleming commented:
@amoedoamorim I just updated the + button size, because after seeing it in person, I made a poor design choice. This aligns the size better with the `<Select` remove button component

brianfleming reviewed (changes requested):
@amoedoamorim I am seeing some issues with the shared feeds:
1. Just adding a collaborator email address triggers validation on the main form even though I have not tried to submit the form:
<img width="1202" alt="Screenshot 2023-10-31 at 9 40 15 AM" src="https://github.com/meedan/check-web/assets/418001/87d4e86b-ba33-4c3b-b272-1ace1a5c1b46">

2. When I go to edit a current shared feed, the header is showing content from the create shared feed view:
<img width="816" alt="Screenshot 2023-10-31 at 9 44 10 AM" src="https://github.com/meedan/check-web/assets/418001/a00617c1-29d5-48dc-831a-1a8444f08bb5">


3. When I create a shared feed correctly, adding a collaborator, but NOT selecting a list I am correctly not being redirected to the new shared feed because there is no list, which is confusing and a design oversight. can we please:
- Change the view so you know you are at least looking at the edit view now
- add a confirmation alert telling the user the shared feed has been created successfully

4. ^^ To go along with that, see video below, one issue is, I am not getting the modal confirming the invitations will be sent.


https://github.com/meedan/check-web/assets/418001/bd128fad-1d4d-4e33-9e17-68e037836911

dariusk commented on src/app/components/feed/FeedCollaboration.js:
```diff
@@ -0,0 +1,284 @@
+/* eslint-disable relay/unused-fields */
```
Is this something we plan to get rid of before merge?

dariusk reviewed (approved):
From a code perspective this all looks good; I left a short question about the `relay/unused-fields` eslint directive.

amoedoamorim commented:
> 1. Just adding a collaborator email address triggers validation on the main form even though I have not tried to submit the form:

This is consistent with any other interaction in the page/form. If we should change it I'd preferably address it in another ticket as there's enough changes and complexity to be a separate ticket.

> 2. When I go to edit a current shared feed, the header is showing content from the create shared feed view:

Changed title when editing

> 3. When I create a shared feed correctly, adding a collaborator, but NOT selecting a list I am correctly not being redirected to the new shared feed because there is no list, which is confusing and a design oversight.

Looks like that was something off there. We should be able to create feeds without selecting a list and be redirected to feed page. That's the current behavior already on live.

> 4. ^^ To go along with that, see video below, one issue is, I am not getting the modal confirming the invitations will be sent.

Fixed confirmation modal!

brianfleming reviewed (changes requested):
Looking great @amoedoamorim ! Thanks for the fixes.
I am only seeing one small thing left before moving on. Very minor but. When you are Creating a new shared feed, and you get the modal about emailing the collaborators, can we update the button to be `Create Shared Feed` instead of `Update Shared Feed`, which should remain only for updates. See screenshot:

<img width="1185" alt="Screenshot 2023-11-09 at 9 38 01 AM" src="https://github.com/meedan/check-web/assets/418001/7fc110e9-3f8b-4de3-a114-9c49a3568d83">

brianfleming commented:
![giphy](https://github.com/meedan/check-web/assets/418001/fb2ad4df-b8ab-4b72-93c1-0991dfdccd64)
